src/main/resources/avro/alleleAnnotationmethods.avdl

```diff
@@ -0,0 +1,200 @@
+annotation of all alleles vs all features in the region is returned. Where a
+set of features is supplied, only annotations against these features (matching
+on featuretype and id) are returned and other overlapping features are ignored.
+
+variantannotationsets/search returns information on the input to the
+annotation. This will be a VariantSet and the reference data and software
+versions used in calculating the annotation.
+It is essential this information is exhaustive.
+*/
+
+/******************  /variantannotations/search  *********************/
+/** 
+This request maps to the body of `POST /variantannotations/search` as JSON
+*/
+record SearchVariantAnnotationsRequest {
+  /** Required. The ID of the variant annotation set to search over. */
+  string variantAnnotationSetId;
+
+  /**
+  Only return variants with reference alleles on the reference with this
+  name. One of this field or `referenceId` or `features` is required.
+  (case-sensitive, exact match)
+  */
+  union { null, string } referenceName = null;
+
+  /**
+  Only return variants with reference alleles on the reference with this
+  ID. One of this field or `referenceName` or `features` is required.
+  */
+  union { null, string } referenceId = null;
+
+  /**
+  Required if referenceName or referenceId supplied.
+  The beginning of the window (0-based, inclusive) for which variants with
+  overlapping reference alleles should be returned.
+  Genomic positions are non-negative integers less than reference length.
+  Requests spanning the join of circular genomes are represented as
+  two requests one on each side of the join (position 0).
+  */
+  long start;
+
+  /**
+  Required if referenceName or referenceId supplied.
+  The end of the window (0-based, exclusive) for which variants with
+  overlapping reference alleles should be returned.
+  */
+  long end;
+
+//  This section will be re-instated when features are available in the API
+//  /**
+//  Only return variant annotations for any of these features.
+//  Features may include specific transcripts or genes. A search by gene will
+//  return information for all transcripts associated with the gene in the
+//  variant annotation set.
+//  This or a location (referenceName/referenceId plus optional start and end)
+//  must be supplied.
+//  If null, return all variant annotations in specified window.
+//  */
+//  union { null, array<string> } featureIds = null;
+
+  /**
+  This filter allows variant, transcript combinations to be extracted by effect
+  type(s).
+  Only return variant annotations including any of these effects and only return
+  transcript effects including any of these effects. Exact matching across all
+  fields of the Sequence Ontology OntologyTerm is required.
+  (A transcript effect may have multiple SO effects which will all be reported.)
+  If null, return all variant annotations.
+  */
+  union { null, array<org.ga4gh.models.OntologyTerm> } effects = null;
```

Comment:
Right now, only the id is required for OntologyTerm.

If this is hard to implement, I would favor delaying
implementation.  I am no sure passing in OntologyTerm is right,
however passing in generic strings isn't right;

I am pushing the metadata task team on refining OntologyTerm; if
you say "we don't know how to implement something", that is
*really* helpful.  Please create tickets and add MetadataTaskTeam
label.



Maciek Smuga-Otto <notifications@github.com> writes:
> @sarahhunt, @diekhans Passing in whole JSON objects representing OntologyTerm
> into a query seems a bit strange. In Sequence Annotations, I replaced this with
> an array<string> ontologyTerms = []; so that a search could just be performed
> on the (presumably standardized) terms.
> 
> I don't know which approach is correct, but I believe we should standardize on
> a single way of searching ontologies across GA4GH API's.
> 
> —
> Reply to this email directly or view it on GitHub.*
>